src/overlay073/ov73_021D0D80.c

```diff
@@ -425,222 +680,260 @@ static void ov73_021D1238(UnkStruct_ov73_021D1058 *param0)
+static BOOL RowanIntro_FadeBgLayer(
+    RowanIntro *manager,
+    int bgLayer,
+    enum FadeDirection fadeDirection)
 {
-    BOOL v0 = 0;
-    GXBlendPlaneMask v1;
-    int v2;
+    BOOL isFinished = FALSE;
+    GXBlendPlaneMask blendPlaneMask;
+    int isSubLayer;
 
-    switch (param1) {
+    switch (bgLayer) {
     default:
-    case 0:
-        v1 = GX_BLEND_PLANEMASK_BG0;
-        v2 = 0;
+    case BG_LAYER_MAIN_0:
+        blendPlaneMask = GX_BLEND_PLANEMASK_BG0;
+        isSubLayer = FALSE;
         break;
-    case 1:
-        v1 = GX_BLEND_PLANEMASK_BG1;
-        v2 = 0;
+    case BG_LAYER_MAIN_1:
+        blendPlaneMask = GX_BLEND_PLANEMASK_BG1;
+        isSubLayer = FALSE;
         break;
-    case 2:
-        v1 = GX_BLEND_PLANEMASK_BG2;
-        v2 = 0;
+    case BG_LAYER_MAIN_2:
+        blendPlaneMask = GX_BLEND_PLANEMASK_BG2;
+        isSubLayer = FALSE;
         break;
-    case 4:
-        v1 = GX_BLEND_PLANEMASK_BG0;
-        v2 = 1;
+    case BG_LAYER_SUB_0:
+        blendPlaneMask = GX_BLEND_PLANEMASK_BG0;
+        isSubLayer = TRUE;
         break;
-    case 5:
-        v1 = GX_BLEND_PLANEMASK_BG1;
-        v2 = 1;
+    case BG_LAYER_SUB_1:
+        blendPlaneMask = GX_BLEND_PLANEMASK_BG1;
+        isSubLayer = TRUE;
         break;
-    case 6:
-        v1 = GX_BLEND_PLANEMASK_BG2;
-        v2 = 1;
+    case BG_LAYER_SUB_2:
+        blendPlaneMask = GX_BLEND_PLANEMASK_BG2;
+        isSubLayer = TRUE;
         break;
     }
 
-    switch (param0->unk_78) {
-    case 0:
-        if (param2 == 0) {
-            param0->unk_7C = 0;
-            param0->unk_80 = 16;
-            param0->unk_78 = 1;
-
-            if (v2 == 0) {
-                G2_SetBlendAlpha(v1, GX_BLEND_PLANEMASK_BG1 | GX_BLEND_PLANEMASK_BG2 | GX_BLEND_PLANEMASK_BG3, param0->unk_7C, param0->unk_80);
+    switch (manager->fadeBgLayerState) {
+    case FBL_STATE_INIT:
+        if (fadeDirection == FADE_IN) {
+            manager->fadeBgLayerCurAlpha = 0;
+            manager->fadeBgLayerCurAlphaInv = 16;
+            manager->fadeBgLayerState = FBL_STATE_FADE_IN;
+
+            if (isSubLayer == FALSE) {
+                G2_SetBlendAlpha(
+                    blendPlaneMask,
+                    GX_BLEND_PLANEMASK_BG1 | GX_BLEND_PLANEMASK_BG2 | GX_BLEND_PLANEMASK_BG3,
+                    manager->fadeBgLayerCurAlpha,
+                    manager->fadeBgLayerCurAlphaInv);
             } else {
-                G2S_SetBlendAlpha(v1, GX_BLEND_PLANEMASK_BG1 | GX_BLEND_PLANEMASK_BG2 | GX_BLEND_PLANEMASK_BG3, param0->unk_7C, param0->unk_80);
+                G2S_SetBlendAlpha(
+                    blendPlaneMask,
+                    GX_BLEND_PLANEMASK_BG1 | GX_BLEND_PLANEMASK_BG2 | GX_BLEND_PLANEMASK_BG3,
+                    manager->fadeBgLayerCurAlpha,
+                    manager->fadeBgLayerCurAlphaInv);
             }
 
-            Bg_ToggleLayer(param1, 1);
+            Bg_ToggleLayer(bgLayer, 1);
```

Comment:
```suggestion
            Bg_ToggleLayer(bgLayer, TRUE);
```
This applies to other instances too

Reply:
I have updated them!